src/codecs/encoding/format/raw_message.rs

```diff
@@ -35,9 +41,12 @@ impl Encoder<Event> for RawMessageSerializer {
     type Error = crate::Error;
 
     fn encode(&mut self, event: Event, buffer: &mut BytesMut) -> Result<(), Self::Error> {
+        let message_key = log_schema().message_key();
+
         let bytes = match event {
             Event::Log(log) => log
-                .get(log_schema().message_key())
+                .get_by_meaning(message_key)
+                .or_else(|| log.get(message_key)) // backward compatibility
```

Comment:
This'll be removed once schema requirements are optionally hooked into the topology. At that point, not enabling the feature means it'll default to always looking for the configured message key, rather than the meaning.